planet/clients/data.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2022 Planet Labs, PBC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+"""Functionality for interacting with the data api"""
+import logging
+import typing
+
+from .. import exceptions
+from ..constants import PLANET_BASE_URL
+from ..http import Session
+from ..models import Paged, Request, Response
+
+BASE_URL = f'{PLANET_BASE_URL}/data/v1/'
+SEARCHES_PATH = '/searches'
+STATS_PATH = '/stats'
+
+STATS_INTERVAL = ('hour', 'day', 'week', 'month', 'year')
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Items(Paged):
+    '''Asynchronous iterator over items from a paged response.'''
+    LINKS_KEY = '_links'
+    NEXT_KEY = '_next'
+    ITEMS_KEY = 'features'
+
+
+class DataClient():
+    """High-level asynchronous access to Planet's data API.
+
+    Example:
+        ```python
+        >>> import asyncio
+        >>> from planet import Session, DataClient
+        >>>
+        >>> async def main():
+        ...     async with Session() as sess:
+        ...         cl = DataClient(sess)
+        ...         # use client here
+        ...
+        >>> asyncio.run(main())
+
+        ```
+    """
+
+    def __init__(self, session: Session, base_url: str = None):
+        """
+        Parameters:
+            session: Open session connected to server.
+            base_url: The base URL to use. Defaults to production data API
+                base url.
+        """
+        self._session = session
+
+        self._base_url = base_url or BASE_URL
+        if self._base_url.endswith('/'):
+            self._base_url = self._base_url[:-1]
+
+    def _searches_url(self):
+        return f'{self._base_url}{SEARCHES_PATH}'
+
+    def _stats_url(self):
+        return f'{self._base_url}{STATS_PATH}'
+
+    def _request(self, url, method, data=None, params=None, json=None):
+        return Request(url, method=method, data=data, params=params, json=json)
+
+    async def _do_request(self, request: Request) -> Response:
+        '''Submit a request and get response.
+
+        Parameters:
+            request: request to submit
+        '''
+        return await self._session.request(request)
+
+    async def quick_search(self,
+                           item_types: typing.List[str],
+                           search_filter: dict,
```

Comment:
```suggestion
                           filter: dict,
```

I think "search" goes without saying?

Reply:
Nevermind! filter conflicts with the name of the built-in function. search_filter is fine.